k9mail/src/androidTest/java/com/fsck/k9/activity/MessageReferenceTest.java

```diff
@@ -124,6 +127,11 @@ public void equalsWithMessageReferenceContainingDifferentUidShouldReturnFalse()
         assertEqualsReturnsFalseSymmetrically(messageReferenceOne, messageReferenceTwo);
     }
 
+    @Test
+    public void hashcodeVerifyEqualsContract() {
+        EqualsVerifier.forClass(MessageReference.class).suppress(Warning.NONFINAL_FIELDS).verify();
```

Comment:
.suppress(Warning.NONFINAL_FIELDS) has been added to suppress complain about the fact that the fields of MessageReference are not declared final.

But, changing any non final field inside, after instantiating, will change  the hashcode, while it will still be the same object.

And so, loading a MessageReference object into a Dictionnary or a Hashset, and then modifying it, will change its key, and make it not findable anymore.

In your opinion, should I suppress this flag ?